File: Word Frequency/Word Frequency.py
import re
from collections import Counter

def get_words(paragraph):
	# Lowercase and remove punctuation
	text = re.sub(r"[,.!]", "", paragraph.lower())
	words = text.split()

	# Count frequencies
	counts = Counter(words)

	# Preserve first occurrence order
	unique_words = []
	for w in words:
		if w not in unique_words:
			unique_words.append(w)

	# Sort: frequency desc, then by first occurrence in text
	sorted_words = sorted(unique_words, key=lambda w: (-counts[w], unique_words.index(w)))

	# Return top 3
	return sorted_words[:3]

File: Word Frequency/Word Frequency.java
import java.util.*;

public class WordFrequency {
	public static List<String> getWords(String paragraph) {
		// Lowercase and remove punctuation
		String text = paragraph.toLowerCase().replaceAll("[,.!]", "");
		String[] words = text.split("\\s+");

		// Count frequencies
		Map<String, Integer> counts = new HashMap<>();
		for (String w : words) counts.put(w, counts.getOrDefault(w, 0) + 1);

		// Preserve first occurrence order
		List<String> uniqueWords = new ArrayList<>();
		for (String w : words) if (!uniqueWords.contains(w)) uniqueWords.add(w);

		// Sort: frequency desc, then by first occurrence
		uniqueWords.sort((a, b) -> {
			int cmp = counts.get(b) - counts.get(a);
			if (cmp != 0) return cmp;
			return Integer.compare(uniqueWords.indexOf(a), uniqueWords.indexOf(b));
		});

		// Return top 3
		return uniqueWords.subList(0, Math.min(3, uniqueWords.size()));
	}

	public static void main(String[] args) {
		System.out.println(getWords("Hello world! Hello, code. Hello code world."));
	}
}
